action.yml

```diff
@@ -9,10 +9,22 @@ runs:
   steps:
     - name: Checkout
       uses: actions/checkout@v3
+    - name: Determine Project Type
+      id: project_type
+      run: |
+        project_type="${{ inputs.project-type }}"
+      shell: bash
+      continue-on-error: true
     - name: Download Detekt Configuration
       run: |
-        echo 'Downloading configuration'
-        curl -L -o detekt.yml https://raw.githubusercontent.com/monta-app/detekt-action/main/conf/detekt.yml
+        project_type="${{ env.PROJECT_TYPE }}"
+        if [ "$project_type" == "ktor" ]; then
+          echo 'Downloading ktor configuration'
+          curl -L -o detekt.yml https://raw.githubusercontent.com/monta-app/detekt-action/main/conf/ktor-detekt.yml
+        else
+          echo 'Downloading micronaut configuration'
+          curl -L -o detekt.yml https://raw.githubusercontent.com/monta-app/detekt-action/main/conf/detekt.yml
```

Comment:
suggestion: Here, we have two different detekt files -- but they are very similar. Over time, there is a danger that they will diverge -- e.g. when we update `detekt.yml`, do we then not also want to (at least consider) whether `ktor-detekt.yml` should also be updated? If that is the case, another approach would be to have a `detekt.yml` as the base, and then have a patch-file, which would be applied for ktor-projects. The result of `detekt.yml` + patch file would then be the resulting detekt-file for ktor. Just something to think about -- it's overkill if the two configs are expected to be very different, but it might be good if they are expected to be similar.

Reply:
that is a good point. I did read about having 2 config files specified, and in case of conflict it will automatically solve it, I'm just not quite sure which file will be overwriting the other. Also the decision for having 2 from the option of having completely different cofigs, but I do not foresee these files being so different, now that I think about it. I will look into it, and test some things :) Thank you for the feedback